libstorage.apib

```diff
@@ -468,33 +531,50 @@ Internal server error
 
     + Schema
 
-            { "$ref": "https://raw.githubusercontent.com/emccode/libstorage/master/libstorage.json#/definitions/internalServerError" }
+            { "$ref": "https://raw.githubusercontent.com/codedellemc/libstorage/master/libstorage.json#/definitions/internalServerError" }
 
 ## Get with Attachments [GET /volumes?{attachments}]
 Gets a list of Volume resources and their attached resources for all configured
 services.
 
 + Parameters
 
-    + attachments (required)
-
-        A flag that indicates that volume attachments should be retrieved.
-        If this flag is provided then the `LibStorage-InstanceID` header must
-        be set -- a comma-delimited list of service name/instance ID,
-        colon-delimited pairs.
+    + attachments (string,optional)
+
+        A value that indicates how and if volume attachments should be retrieved.
+        <br/><br/>
+        The `attachments` parameter can be any of the following values:
+        <br/>
+        Value | Description
+        ------|------------
+        0 | No attachment information is requested. This is the default value and the same as omitting this parameter altogether.
+        1 | Attachment information is requested. This value indicates attachment information should be provided for all volumes.
+        2 | _My_ attachment information is requested. This value indicates attachment information should be provided for volumes attached to the instance specified in the [instance ID](#def-instance-id) request header should be returned. If this bit is set then the instance ID header is required.
+        4 | Attempt to map devices provided via the [local devices](#def-local-devices) request header to the appropriate attachment information. If this bit is set then the instance ID and local device headers are required.
```

Comment:
> the client can just send the header multiple times for each service

Technically the client sends the header multiple times, one per unique driver type. For example, imagine a libStorage server configured as such:

```yaml
libstorage:
  server:
    services:
      ebs00:
        driver: ebs
      ebs01:
        driver: ebs
      ebs02:
        driver: ebs
      sio00:
        driver: scaleio
      sio01:
        driver: scaleio
```

The above libStorage configuration defines five services: three using the EBS driver and two using the ScaleIO driver. However, when the libStorage client sends the server instance ID and local device headers, this is what will be sent:

```
Libstorage-Instanceid: ebs=i-2348,region%3Dwest
Libstorage-Instanceid: scaleio=4f580-3091
Libstorage-Localdevices: ebs=/dev/xvdf::/dev/xvdf
Libstorage-Instanceid: scaleio=vol-123::/dev/scini1,vol-349::/dev/scini2
```

Even though the server has three configured services that use the EBS driver and two configured services that use the ScaleIO driver, the client only needs to send the instance ID and local devices for EBS once. The same for ScaleIO. 

The use of multiple headers correlates to unique drivers for configured services, not for the number of configured services themselves.

Does that make sense?